src/psyclone/tests/psyir/frontend/fparser2_pointer_test.py

```diff
@@ -58,8 +58,28 @@ def test_pointer_assignments(fortran_reader):
     end subroutine
     '''
     file_container = fortran_reader.psyir_from_source(test_module)
-    assert len(file_container.walk(CodeBlock)) == 0
+    assert not file_container.walk(CodeBlock)
     assignments = file_container.walk(Assignment)
     assert len(assignments) == 2
     for assignment in assignments:
         assert assignment.is_pointer is True
+
+
+def test_unsupported_pointer_assignments(fortran_reader):
+    '''
+    Test that pointer assignments that have and array accessor syntax on the
+    inner element are not supported.
+    '''
+    test_module = '''
+    subroutine mysub()
+        use other_symbols
+        integer, target :: a = 1
+        integer, pointer :: b => null()
```

Comment:
Strictly, shouldn't `b` be of `dimension(3)` or larger?

Reply:
Ah yes, but I removed the declarations (are imported) since I want to focus on the expressions here. I also added an array of pointers assignment that is not in a structure.